### PR TITLE
FLIP - Flow Client Library (FCL) Wallet Provider Specification

### PR DESCRIPTION
**Pull request by gregsantos**
This FLIP proposes a v1.0 specification for FCL Wallet Providers and dApps to trustlessly interact for the purposes of authentication, and transaction and message signing on Flow. It aims to provide a comprehensive description of the communication protocol and the required features and functionality expected to perform these functions.

### CODE REVIEW

File: application/20221108-fcl-wallet-provider-spec.md [hunk 1]
```diff
@@ -0,0 +1,1168 @@
...
+| Status        | Proposed                                    |
+| :------------ | :------------------------------------------ |
+| **FLIP #**    | [NNN](Link to FLIP)                         |
+| **Author(s)** | Greg Santos (greg.santos@dapperlabs.com)    |
+| **Sponsor**   | Justin Barry (justin.barry@dapperlabs.com ) |
+| **Updated**   | 2022-11-08                                  |
+
+## Objective
+
+This FLIP proposes a v1.0 specification for FCL Wallet Providers and dApps to trustlessly interact for the purposes of authentication, and transaction and message signing on Flow. It aims to provide a comprehensive description of the communication protocol and the required features and functionality expected to perform these functions.
```
Comment (srinjoyc) [inline application/20221108-fcl-wallet-provider-spec.md:12]:
```suggestion
This FLIP proposes a v1.0 specification for FCL Wallet Providers and dApps to trustlessly interact for the purposes of authentication and producing signatures on Flow. It aims to provide a comprehensive description of the communication protocol and the required features and functionality expected to perform these functions.
```

File: application/20221108-fcl-wallet-provider-spec.md [hunk 2]
```diff
@@ -0,0 +1,1168 @@
...
+
+The goal of this FLIP is to solidify existing implementation patterns and data definitions in a generalized way. Additionally it should open discussion to potential improvements for future versions of the spec in order to simplify and support Flow dApp and wallet development.
```
Comment (srinjoyc) [inline application/20221108-fcl-wallet-provider-spec.md:14]:
```suggestion
The goal of this FLIP is to solidify existing implementation patterns and data definitions in a generalized way. Additionally it should open discussion to potential improvements for future versions of the spec in order to evolve the Flow dApp and wallet ecosystem.
```

File: application/20221108-fcl-wallet-provider-spec.md [hunk 3]
```diff
@@ -0,0 +1,1168 @@
...
+
+##### <a id="identity"></a> `Identity`
+
+This object is used to define the identity of the user.
+
+```typescript
+interface Identity extends ObjectBase {
+  f_type: "Identity"
+  address: string
+  keyId?: number
```
Comment (justinbarry) [inline application/20221108-fcl-wallet-provider-spec.md:343]: Is this required for the proposer on preauthz response? cc: @JeffreyDoyle

Comment (JeffreyDoyle) [reply application/20221108-fcl-wallet-provider-spec.md]: Yes this is for the proposer!

**Review by justinbarry — APPROVED** [review]
LGTM

**Pull request merged**